File: Python-Basics/day14.py
def expense_tracker():
    # Ek khali list jisme saara data store hoga
    products = []
    
    print("--- Welcome to Product Expense Tracker ---")
    
    while True:
        # User se product ka naam lena
        product_name = input("\nProduct ka naam daalein (ya 'exit' likhein band karne ke liye): ").strip()
        
        # Agar user exit likhe toh loop band ho jaye
        if product_name.lower() == 'exit':
            break
            
        # Price (Kharcha) input lena aur handle karna agar galat number daale
        try:
            price = float(input(f"'{product_name}' ka kharcha (Price) daalein: ₹"))
        except ValueError:
            print("❌ Kripya sahi number (amount) daalein!")
            continue
            
        # Product aur price ko dictionary me daal kar list me save karna
        products.append({"name": product_name, "price": price})
        print(f"✅ '{product_name}' save ho gaya!")

    # --- Ab Final Hisaab Show Karte Hain ---
    print("\n" + "="*30)
    print("📋 AAPKA FINAL HISAAB KITAB")
    print("="*30)
    
    if not products:
        print("Aapne koi kharcha add nahi kiya.")
    else:
        total_expense = 0
        # Saare products ko ek ek karke print karna
        for index, item in enumerate(products, 1):
            print(f"{index}. {item['name']}: ₹{item['price']:.2f}")
            total_expense += item['price']
            
        print("-"*30)
        print(f"💰 KUL KHARCHA (Total Expense): ₹{total_expense:.2f}")
        print("="*30)

# Program ko chalane ke liye function call karein
expense_tracker()

File: CPP-Basics/day14.cpp
#include <iostream>
#include <vector>
#include <string>
#include <iomanip> // Price ko .2f format me dikhane ke liye

using namespace std;

// Product ke liye ek structure (blueprint)
struct Product {
    string name;
    double price;
};

int main() {
    vector<Product> products; // Saare products store karne ke liye list
    string name;
    double price;

    cout << "--- Welcome to Product Expense Tracker (C++) ---" << endl;

    while (true) {
        cout << "\nProduct ka naam daalein (ya 'exit' likhein band karne ke liye): ";
        cin >> name;

        // Agar user exit likhe toh loop band ho jaye
        if (name == "exit" || name == "EXIT") {
            break;
        }

        cout << "'" << name << "' ka kharcha (Price) daalein: ";
        cin >> price;

        // Input validation: Agar user number ki jagah text daal de
        if (cin.fail()) {
            cout << "❌ Kripya sahi number (amount) daalein!" << endl;
            cin.clear(); // Error flag clear karne ke liye
            cin.ignore(10000, '\n'); // Galat input ko skip karne ke liye
            continue;
        }

        // New product ko vector (list) mein add karna
        products.push_back({name, price});
        cout << "✅ '" << name << "' save ho gaya!" << endl;
    }

    // --- Ab Final Hisaab Show Karte Hain ---
    cout << "\n==============================" << endl;
    cout << "📋 AAPKA FINAL HISAAB KITAB" << endl;
    cout << "==============================" << endl;

    if (products.empty()) {
        cout << "Aapne koi kharcha add nahi kiya." << endl;
    } else {
        double total_expense = 0;

        // Price ko hamesha 2 decimal tak dikhane ke liye setting
        cout << fixed << setprecision(2);

        // Saare products ko loop se print karna
        for (size_t i = 0; i < products.size(); ++i) {
            cout << i + 1 << ". " << products[i].name << ": " << products[i].price << " Rs" << endl;
            total_expense += products[i].price;
        }

        cout << "------------------------------" << endl;
        cout << "💰 KUL KHARCHA (Total Expense): " << total_expense << " Rs" << endl;
        cout << "==============================" << endl;
    }

    return 0;
}